compass/Coordinator.java

```diff
@@ -132,7 +162,7 @@ protected boolean nodeIsSolid(GetNodeInfoResponse nodeInfo) {
     if (nodeInfo.getLatestSolidSubtangleMilestoneIndex() != nodeInfo.getLatestMilestoneIndex())
       return false;
 
-    if (!config.inception && (nodeInfo.getLatestSolidSubtangleMilestoneIndex() != latestMilestone))
+    if (!config.inception && (nodeInfo.getLatestSolidSubtangleMilestoneIndex() != state.latestMilestoneIndex))
```

Comment:
Whoops, good thing I am reading it again and again

`state.latestMilestoneIndex` is `Integer` not `int`
If `nodeInfo.getLatestSolidSubtangleMilestoneIndex()` also returns `Integer` (even if it returns `int` it may change in the future) then we are checking for objects inequality.

This means that 2 Integers that are not in the java integer pool [-128,127] may not be equal even if they have the same nominal value

Reply:
Yeah, `nodeInfo.getLatestSolidSubtangleMilestoneIndex()` returns `int`. I am changing `latestMilestoneIndex` to `int` too to have the same type and avoid awkward casts.

Reply:
as long as the serialization still works fine :-)
If it doesn't (I don't remember if you must have an Integer) you can also use the `equals()` method

Reply:
Serialization also saves the type of the object being saved, so there is no way the serialization or deserialization processes will magically cast or round types.

Reply:
:+1:
The only thing is that with ints we can't have `null` value, but I think this is fine for our case